diff -Naur --exclude '.*' --exclude '*.pyc' --exclude '*.orig' --exclude '*.rej' gfx/cairo/cairo/src/moz.build gfx/cairo-patched/cairo/src/moz.build
--- gfx/cairo/cairo/src/moz.build	2016-11-14 10:22:06.000000000 +0100
+++ gfx/cairo-patched/cairo/src/moz.build	2018-03-23 05:58:07.000000000 +0100
@@ -23,14 +23,7 @@
     ]
     SOURCES += [
         'cairo-base85-stream.c',
-        'cairo-cff-subset.c',
-        'cairo-deflate-stream.c',
-        'cairo-pdf-operators.c',
-        'cairo-pdf-surface.c',
-        'cairo-truetype-subset.c',
         # cairo-type1-subset.c should be here, but it's only supported on freetype platforms
-        'cairo-type1-fallback.c',
-        'cairo-type3-glyph-surface.c',
     ]
 
 if CONFIG['MOZ_WIDGET_TOOLKIT'] == 'windows':
@@ -119,10 +112,12 @@
     'cairo-botor-scan-converter.c',
     'cairo-boxes.c',
     'cairo-cache.c',
+	'cairo-cff-subset.c',
     'cairo-clip.c',
     'cairo-color.c',
     'cairo-composite-rectangles.c',
     'cairo-debug.c',
+	'cairo-deflate-stream.c',
     'cairo-device.c',
     'cairo-fixed.c',
     'cairo-font-face-twin-data.c',
@@ -150,7 +145,10 @@
     'cairo-path-stroke.c',
     'cairo-path.c',
     'cairo-pattern.c',
+	'cairo-pdf-operators.c',
+	'cairo-pdf-surface.c',
     'cairo-pen.c',
+	'cairo-png.c',
     'cairo-polygon.c',
     'cairo-recording-surface.c',
     'cairo-rectangle.c',
@@ -168,10 +166,14 @@
     'cairo-surface-snapshot.c',
     'cairo-surface-subsurface.c',
     'cairo-surface.c',
+	'cairo-svg-surface.c',
     'cairo-tee-surface.c',
     'cairo-tor-scan-converter.c',
     'cairo-toy-font-face.c',
     'cairo-traps.c',
+	'cairo-truetype-subset.c',
+	'cairo-type1-fallback.c',
+	'cairo-type3-glyph-surface.c',
     'cairo-unicode.c',
     'cairo-user-font.c',
     'cairo-version.c',
