diff -Naur --exclude '.*' --exclude '*.pyc' gfx/2d/Factory.cpp gfx/2d-patched/Factory.cpp
--- gfx/2d/Factory.cpp	2016-08-01 09:12:55.000000000 +0200
+++ gfx/2d-patched/Factory.cpp	2016-10-15 20:58:58.000000000 +0200
@@ -14,6 +14,12 @@
 #ifdef USE_SKIA
 #include "DrawTargetSkia.h"
 #include "ScaledFontBase.h"
+#if USE_SKIA_GPU
+#include "gfxPlatform.h"
+#include "skia/include/gpu/GrContext.h"
+#include "skia/include/gpu/gl/GrGLInterface.h"
+#include "SkiaGLGlue.h"
+#endif
 #ifdef MOZ_ENABLE_FREETYPE
 #define USE_SKIA_FREETYPE
 #include "ScaledFontCairo.h"
@@ -335,14 +341,29 @@
 #endif
 #ifdef USE_SKIA
   case BackendType::SKIA:
-    {
-      RefPtr<DrawTargetSkia> newTarget;
-      newTarget = new DrawTargetSkia();
-      if (newTarget->Init(aSize, aFormat)) {
-        retVal = newTarget;
-      }
-      break;
-    }
+  {
+
+#if USE_SKIA_GPU
+#include "gfxPlatform.h"
+	  gl::SkiaGLGlue* glue = gfxPlatform::GetPlatform()->GetSkiaGLGlue();
+
+	  if (glue && glue->GetGrContext() && glue->GetGLContext()) {
+		  RefPtr<DrawTarget> newTarget;
+		  newTarget = Factory::CreateDrawTargetSkiaWithGrContext(glue->GetGrContext(), aSize, aFormat);
+		  if (newTarget) {
+			  retVal = newTarget;
+		  } else {
+			  RefPtr<DrawTargetSkia> newTarget;
+			  newTarget = new DrawTargetSkia();
+			  if (newTarget->Init(aSize, aFormat)) {
+				  retVal = newTarget;
+			  }
+		  }
+	  }
+#endif
+
+	  break;
+  }
 #endif
 #ifdef USE_CAIRO
   case BackendType::CAIRO: 
