diff -Naur --exclude '.*' --exclude '*.pyc' --exclude '*.orig' --exclude '*.rej' gfx/src/DriverCrashGuard.cpp gfx/src-patched/DriverCrashGuard.cpp
--- gfx/src/DriverCrashGuard.cpp	2016-11-14 10:22:06.000000000 +0100
+++ gfx/src-patched/DriverCrashGuard.cpp	2017-01-10 10:30:09.000000000 +0100
@@ -106,8 +106,7 @@
   if (XRE_IsContentProcess()) {
     // Ask the parent whether or not activating the guard is okay. The parent
     // won't bother if it detected a crash.
-    dom::ContentChild* cc = dom::ContentChild::GetSingleton();
-    cc->SendBeginDriverCrashGuard(uint32_t(mType), &mCrashDetected);
+    mCrashDetected = false;
     if (mCrashDetected) {
       LogFeatureDisabled();
       return;
@@ -162,7 +161,6 @@
       SetStatus(DriverInitStatus::Okay);
     }
   } else {
-    dom::ContentChild::GetSingleton()->SendEndDriverCrashGuard(uint32_t(mType));
   }
 
 #ifdef MOZ_CRASHREPORTER
diff -Naur --exclude '.*' --exclude '*.pyc' --exclude '*.orig' --exclude '*.rej' gfx/src/nsScriptableRegion.cpp gfx/src-patched/nsScriptableRegion.cpp
--- gfx/src/nsScriptableRegion.cpp	2016-11-14 10:22:06.000000000 +0100
+++ gfx/src-patched/nsScriptableRegion.cpp	2017-01-10 10:30:09.000000000 +0100
@@ -136,24 +136,5 @@
     return NS_OK;
   }
 
-  JS::Rooted<JSObject*> destArray(aCx, JS_NewArrayObject(aCx, numRects * 4));
-  if (!destArray) {
-    return NS_ERROR_OUT_OF_MEMORY;
-  }
-
-  aRects.setObject(*destArray);
-
-  uint32_t n = 0;
-  for (auto iter = mRegion.RectIter(); !iter.Done(); iter.Next()) {
-    const mozilla::gfx::IntRect& rect = iter.Get();
-    if (!JS_DefineElement(aCx, destArray, n, rect.x, JSPROP_ENUMERATE) ||
-        !JS_DefineElement(aCx, destArray, n + 1, rect.y, JSPROP_ENUMERATE) ||
-        !JS_DefineElement(aCx, destArray, n + 2, rect.width, JSPROP_ENUMERATE) ||
-        !JS_DefineElement(aCx, destArray, n + 3, rect.height, JSPROP_ENUMERATE)) {
-      return NS_ERROR_FAILURE;
-    }
-    n += 4;
-  }
-
   return NS_OK;
 }
